server/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -110,6 +110,8 @@ default Repository create(RepositoryMetadata metadata, Function<String, Reposito
      * Returns a {@link RepositoryData} to describe the data in the repository, including the snapshots
      * and the indices across all snapshots found in the repository.  Throws a {@link RepositoryException}
      * if there was an error in reading the data.
+     * @param listener listener that may be resolved on different kinds of threads including transport and cluster state applier threads
```

Comment:
I audited all usages of this method and I couldn't find any remaining spot where this is a problem (at least in `master`, might need some adjustments in `7.x`).